src/BillWidget.h

```diff
@@ -0,0 +1,135 @@
+#   Author:             stwomey
+#   Description:        Header file for a BillWidget
+#   Change History:     See https://github.com/SeanTwomey9/PersonalFinanceTool
+#   Relative Location:  PersonalFinanceTool/src/
+##################################################################################*/
+
+#ifndef BILLWIDGET_H
+#define BILLWIDGET_H
+
+#include <QDialog>
+#include <QLabel>
+#include <QLineEdit>
+#include <QDateEdit>
+#include <QGridLayout>
+#include <QPushButton>
+
+/**
+ * @brief Properly deletes a dynamically allocated label pointer, as long as it was created properly.
+ * @param p_label - The label pointer to be deleted.
+ */
+void deleteLabelIfNonNull(QLabel *p_label);
+
+/**
+ * @brief Properly deletes a dynamically allocated line edit pointer, as long as it was created properly.
+ * @param p_lineEdit - The line edit pointer to be deleted.
+ */
+void deleteLineEditIfNonNull(QLineEdit *p_lineEdit);
+
+/**
+ * @brief Properly deletes a dynamically allocated button pointer, as long as it was created properly.
+ * @param p_button - The button pointer to be deleted.
+ */
+void deleteButtonIfNonNull(QPushButton *p_button);
+
+/**
+ * @brief The BillWidget class represents a pop-up widget which allows the user to enter a bill that they wish to track.
+ */
+class BillWidget : public QWidget
+{
+    Q_OBJECT
+
+public:
+
+    /**
+     * @brief Constructs a widget allowing the user to enter their bills.
+     * @param parent - Parent widget used for object tree tracking.
+     */
+    BillWidget(QObject *parent);
+
+    /**
+     * @brief Destructs the BillWidget by deleting objects which were dynamically allocated.
+     */
+    ~BillWidget();
+
+    /**
+     * @brief Creates the labels, input fields, and buttons of the BillWidget.
+     */
+    void initializeWidget();
+
+    /**
+     * @brief Adds the widgets of the BillWidget into a grid layout.
+     */
+    void createWidgetLayout();
+
+    /**
+     * @brief Returns the close push button pointer.
+     * @return The close push button pointer.
+     */
+    QPushButton* getCloseButton();
+
+    /**
+     * @brief Returns the enter another push button pointer.
+     * @return The enter another push button pointer.
+     */
+    QPushButton* getEnterAnotherButton();
+
+    /**
+     * @brief Returns the done push button pointer.
+     * @return The done push button pointer.
+     */
+    QPushButton* getDoneButton();
+
+    /**
+     * @brief Returns the name line edit pointer.
+     * @return The name line edit pointer.
+     */
+    QLineEdit* getNameInput();
+
+    /**
+     * @brief Returns the amount due line edit pointer.
+     * @return The amount due line edit pointer.
+     */
+    QLineEdit* getAmountDueInput();
+
+    /**
+     * @brief Returns the due date date edit pointer.
+     * @return The due date date edit pointer.
+     */
+    QDateEdit* getDueDateInput();
+
+private:
+
+
+    const QString m_WINDOW_TITLE = "Bill Entry"; //!< Title of the BillWidget window.
```

Comment:
Try to organize `m_WINDOW_TITLE` and `m_gridLayout` into one of the commented groupings so they're not isolated.

Reply:
Variables organized.